# Exercise 7


In [7]:
import polars as pl

df = pl.read_csv('../data/migraine.csv')

# Print contingency table:
df.pivot('pain_free', 'group', 'pain_free', 'count', sort_columns=True)


shape: (2, 3)
┌───────────┬─────┬─────┐
│ group     ┆ no  ┆ yes │
│ ---       ┆ --- ┆ --- │
│ str       ┆ u32 ┆ u32 │
╞═══════════╪═════╪═════╡
│ treatment ┆ 33  ┆ 10  │
│ control   ┆ 44  ┆ 2   │
└───────────┴─────┴─────┘

In [18]:
df.filter(pl.col('group') == 'treatment').group_by('pain_free').agg(pl.count().alias('count')).with_columns((pl.col('count') * 100 / pl.sum('count')).round(2))

shape: (2, 2)
┌───────────┬───────┐
│ pain_free ┆ count │
│ ---       ┆ ---   │
│ str       ┆ f64   │
╞═══════════╪═══════╡
│ yes       ┆ 23.26 │
│ no        ┆ 76.74 │
└───────────┴───────┘

In [17]:
df.filter(pl.col('group') == 'control').group_by('pain_free').agg(pl.count().alias('count')).with_columns((pl.col('count') * 100 / pl.sum('count')).round(2))

shape: (2, 2)
┌───────────┬───────┐
│ pain_free ┆ count │
│ ---       ┆ ---   │
│ str       ┆ f64   │
╞═══════════╪═══════╡
│ yes       ┆ 4.35  │
│ no        ┆ 95.65 │
└───────────┴───────┘

Hence, 23.26% of the treatment group and %4.35 of the control group were pain free after 24 hrs.

This is not sufficient to conclude that acupunture is a useful treatment. The results could be down to chance.

#### Vars
* Explanatory: Acupuncture treatment (y/n)
* Response: Pain Free (y/n)